roles/foreman_staging_repositories/tasks/redhat.yml

```
@@ -0,0 +1,98 @@
+---
+- name: 'Remove foreman-release'
+  yum:
+    name: foreman-release
+    state: absent
+
+- name: 'Create module_hotfixes config file'
+  copy:
+    dest: /etc/yum/foreman.conf
+    content: |
+      module_hotfixes=1
+
+- name: 'Foreman {{ foreman_staging_repositories_version }} Koji repository'
+  yum_repository:
+    name: foreman-koji
+    description: "Foreman {{ foreman_staging_repositories_version }} Koji Repository"
+    baseurl: "http://koji.katello.org/releases/yum/foreman-{{ foreman_staging_repositories_version }}/RHEL/{{ ansible_distribution_major_version }}/x86_64/"
+    priority: "1"
+    gpgcheck: no
+    include: /etc/yum/foreman.conf
+
+- name: 'Foreman {{ foreman_staging_repositories_version }} Plugins Koji repository'
+  yum_repository:
+    name: foreman-plugins-koji
+    state: "present"
+    description: "Foreman Plugins {{ foreman_staging_repositories_version }} Koji Repository"
+    baseurl: "http://koji.katello.org/releases/yum/foreman-plugins-{{ foreman_staging_repositories_version }}/RHEL/{{ ansible_distribution_major_version }}/x86_64/"
+    priority: "1"
+    gpgcheck: no
+    include: /etc/yum/foreman.conf
+
+- when: foreman_staging_repositories_katello_version is defined
+  block:
+    - name: 'Remove katello-repos'
+      yum:
+        name: katello-repos
+        state: absent
+
+    - name: 'Katello {{ foreman_staging_repositories_katello_version }} Koji repository'
+      yum_repository:
+        name: katello-koji
+        description: "Katello {{ foreman_staging_repositories_katello_version }} Koji Repository"
+        baseurl: "http://koji.katello.org/releases/yum/katello-{{ foreman_staging_repositories_katello_version }}/katello/el{{ ansible_distribution_major_version }}/x86_64/"
+        priority: '1'
+        gpgcheck: no
+        include: /etc/yum/foreman.conf
+
+    - name: 'Candlepin Koji repository'
+      yum_repository:
+        name: candlepin-koji
+        description: "Candlepin {{ foreman_staging_repositories_katello_version }} Koji Repository"
+        baseurl: "http://koji.katello.org/releases/yum/katello-{{ foreman_staging_repositories_katello_version }}/candlepin/el{{ ansible_distribution_major_version }}/x86_64/"
+        priority: '1'
+        gpgcheck: no
+        include: /etc/yum/foreman.conf
+
+    - name: 'Add Pulp {{ foreman_staging_repositories_pulp_version }} Stable repository'
+      yum_repository:
+        name: pulp-repository
+        description: Pulp release repository
+        baseurl: "https://repos.fedorapeople.org/repos/pulp/pulp/{{ foreman_staging_repositories_pulp_release }}/{{ foreman_staging_repositories_pulp_version }}/$releasever/$basearch"
+        gpgcheck: no
+        enabled: yes
+      when:
+        - foreman_staging_repositories_katello_version is version('4.0', '<=')
```

This will fail if `foreman_staging_repositories_katello_version` is `nightly`

```suggestion
        - foreman_staging_repositories_katello_version != 'nightly'
        - foreman_staging_repositories_katello_version is version('4.0', '<=')
```

---

addendum: fail on python3, not on 2 :woman_shrugging: